libkernelflinger/security.c

```diff
@@ -216,6 +233,107 @@ struct rot_data_t* get_rot_data()
 	return &rot_data;
 }
 
+/* Update the struct attestation_ids for startup_information */
+EFI_STATUS update_attestation_ids(IN VOID *vendorbootimage)
+{
+    EFI_STATUS ret = EFI_SUCCESS;
+
+    struct vendor_boot_img_hdr_v4 *vendor_hdr = (struct vendor_boot_img_hdr_v4 *)vendorbootimage;
+    UINT32 page_size = vendor_hdr->page_size;
+    UINT32 bootconfig_offset = ALIGN(sizeof(struct vendor_boot_img_hdr_v4), page_size) +
+                               ALIGN(vendor_hdr->vendor_ramdisk_size, page_size) +
+                               ALIGN(vendor_hdr->dtb_size, page_size) +
+                               ALIGN(vendor_hdr->vendor_ramdisk_table_size, page_size);
+    /* Initialize the attestation ids structure */
+    if (vendor_hdr->bootconfig_size) {
```

Comment:
a better coding style can be:
if (vendor_hdr->bootconfig_size < 0)
     return fail;

// the logic code of set the attestation_ids.
xxx;